draft-ietf-tls-tls13.md

```diff
@@ -2735,50 +2747,55 @@ EndOfEarlyData message.
+  which the ticket was established.
+
+The server can determine the client's view of the age of the ticket by
+subtracting the ticket's "ticket_age_add value" from the
+"obfuscated_ticket_age" parameter in the client's "pre_shared_key"
+extension. The server can independently determine its view of the
+age of the ticket by subtracting the the time the ticket was issued
+from the curren time. If the client and server clocks were running
+at the same rate, the client's view of would be shorter than the
 actual time elapsed on the server by a single round trip time.  This
 difference is comprised of the delay in sending the NewSessionTicket
 message to the client, plus the time taken to send the ClientHello to
-the server.  For this reason, a server SHOULD measure the round trip
-time prior to sending the NewSessionTicket message and account for
-that in the value it saves.
+the server.
 
-To properly validate the ticket age, a server needs to save at least two items:
+The mismatch between the client's and server's views of age is thus
+given by:
 
-- The time that the server generated the session ticket and the estimated round
-  trip time can be added together to form a baseline time.
-- The "ticket_age_add" parameter from the NewSessionTicket is needed to recover
-  the ticket age from the "obfuscated_ticket_age" parameter.
+~~~~
+    mismatch = (client's view + RTT estimate) - (server's view)
+~~~~
 
 There are several potential sources of error that make an exact
-measurement of time difficult.  Variations in client and server clock rates
-are likely to be minimal, though potentially with gross time corrections.  Network
-propagation delays are the most likely causes of a mismatch in legitimate
-values for elapsed time.  Both the NewSessionTicket and ClientHello
-messages might be retransmitted and therefore delayed, which might be
-hidden by TCP.
-
-A small allowance for errors in clocks and variations in measurements
-is advisable.  However, any allowance also increases the opportunity
-for replay.  In this case, it is better to reject early data and fall back
-to a full 1-RTT handshake than to risk greater exposure to replay attacks.
-In common network topologies for browser clients, small allowances on the
-order of ten seconds are reasonable.  Clock skew distributions are not
-symmetric, so the optimal tradeoff may involve an asymmetric replay window.
-
+measurement of time difficult. Variations in client and server clock
+rates are likely to be minimal, though potentially with gross time
+corrections.  Network propagation delays are the most likely causes of
+a mismatch in legitimate values for elapsed time.  Both the
+NewSessionTicket and ClientHello messages might be retransmitted and
+therefore delayed, which might be hidden by TCP. Thus, an
+an allowance on the order of ten seconds to account for errors in clocks and
+variations in measurements is advisable. Outside this range, the
```

Comment:
When I wrote the original text, I was careful to scope the numerical guidance to the scenario used to compute it.  It's unclear that 10 seconds is still good for the gamut from IoT to intra-datacenter to space missions.  The RTT measurement helps some, but is not a guarantee.